test/tactics.lean

```diff
@@ -279,6 +279,15 @@ begin
   all_goals { admit }
 end
 
+-- test lift of functions
+example (α : Type*) (f : α → ℤ) (hf : ∀ a, 0 ≤ f a) (hf' : ∀ a, f a < 1) (a : α) : 0 ≤ 2 * f a :=
+begin
+  lift f to α → ℕ using hf,
+    guard_target ((0:ℤ) ≤ 2 * (λ i : α, (f i : ℤ)) a),
```

Comment:
Ideally `lift` would beta reduce things like this. It would be more natural to see `(0:ℤ) ≤ 2 * (f a : ℤ)`. I know `lift` calls `dsimp` in a few places -- maybe this is just a matter of adjusting config options somewhere? But I'm not sure and don't have time right now to try myself.

Reply:
@fpvandoorn You wrote `lift.lean`. Could you please help here? The problem is that I know nothing about all these (greek letter)-reductions (OK, probably I saw a list of definitions somewhere).

Reply:
On second thought, I'm not sure this is so easy. There's not a convenient way to beta reduce applications of only one particular function. And `lift` doesn't consider the applications, just the function term itself. It would be simple enough to beta reduce the whole expression. Would that be confusing?

Reply:
I think that a user may want to call some sort of `simp` / `norm_cast` anyway, and it'll handle beta-reductions.

Reply:
It would be possible to beta-reduce the target and all hypotheses. But as urkud said, usually after applying `lift` you call some kind of simplification procedure to simplify the goal/hypotheses anyway.

Reply:
Okay. Let's trust that this won't be an issue in practice.